### PR TITLE
cranelift: expand umbrella crate with more crates

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Closes https://github.com/bytecodealliance/wasmtime/issues/7772

This adds the `interpreter`, `jit`, `module`, `native` & `object` crates to the umbrella crate. All crates (including the existing `codegen` and `frontend` crates) are feature gated. However, `codegen` and `frontend` are default features, so backwards compatibility is retained if default features is not set to false. I wasn't quite sure which crates to include, but in my own project, I already depend on `jit`, `module` and `native`. Based on their descriptions, `interpreter` and `object` also seemed relevant as public API. I might have missed some other important crate.

I figured this would be an easy fix, although I understand if the umbrella crate should not have all these crates. In that case, feel free to close this PR :)

### CODE REVIEW

File: cranelift/umbrella/Cargo.toml [hunk 1]
```diff
@@ -12,10 +12,30 @@ keywords = ["compile", "compiler", "jit"]
 edition.workspace = true
 
 [dependencies]
-cranelift-codegen = { workspace = true }
-cranelift-frontend = { workspace = true }
+cranelift-codegen = { workspace = true, optional = true }
```
Comment (bjorn3) [inline cranelift/umbrella/Cargo.toml:15]: All other crates depend on cranelift-codegen, so I don't see any reason why you should be able to disable it.

Comment (tertsdiepraam) [reply cranelift/umbrella/Cargo.toml]: Makes sense! Updated!

**Review by elliottt — APPROVED** [review]
This all seems reasonable to me, thanks for fixing this!

**Comment by elliottt** [comment]
The publish step is failing because `cranelift-jit` has a dev-dependency on `cranelift` for its examples. Can we move the jit examples out of `cranelift-jit` to break this cycle? Perhaps the umbrella crate would be a better place to centralize stuff like this now.

**Comment by elliottt** [comment]
(@alexcrichton mentioned that the cycle could be broken by using a `{ path = ... }` version instead, so I went ahead and pushed that change. I'll merge this once it passes branch CI)

**Pull request merged**